src/System.Private.Xml/src/System/Xml/Serialization/ReflectionXmlSerializationWriter.cs

```diff
@@ -477,7 +477,7 @@ private void WriteStructMethod(StructMapping mapping, string n, string ns, objec
             {
                 if (mapping.TypeDesc.Type != null && typeof(XmlSchemaObject).IsAssignableFrom(mapping.TypeDesc.Type))
                 {
-                    throw new PlatformNotSupportedException(typeof(XmlSchemaObject).ToString());
+                    EscapeName = false;
```

Comment:
XmlSchemaObject is abstract. Is it possible that an external derived type requires EscapeName to be set to true?